### PR TITLE
mach: check Platform functions declaration & add deinit for wasm

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

**Comment by slimsag** [comment]
Fixes #556

Fixes #536

File: README.md [hunk 1]
```diff
@@ -76,10 +76,10 @@ Mach works with the new self-hosted Zig compiler only. This means you'll need to
 
 Currently we test using `0.10.0-dev.3952+9e070b653`. Newer is generally better, but you can download this version here:
```
Comment (slimsag) [inline README.md:77]: Hm, I should remove this line. Looks like the script/regex doesn't update it correctly.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -61,7 +61,7 @@ jobs:
       - name: Setup Zig
         run: |
           $ProgressPreference = 'SilentlyContinue'
-          Invoke-WebRequest -Uri 'https://ziglang.org/builds/zig-windows-x86_64-0.10.0-dev.3952+9e070b653.zip' -OutFile 'C:\zig.zip'
+          Invoke-WebRequest -Uri 'https://ziglang.org/builds/zig-windows-x86_64-0.10.0-dev.4060+61aaef0b0.zip' -OutFile 'C:\zig.zip'
           cd C:\
           7z x zig.zip
           Add-Content $env:GITHUB_PATH 'C:\zig-windows-x86_64-0.10.0-dev.3027+0e26c6149\'
```
Comment (alichraghi) [inline .github/workflows/ci.yml:67]: same here. regex didn't update this

Comment (slimsag) [reply .github/workflows/ci.yml:67]: Got it, I'll see if I can sort this out

**Review by slimsag — APPROVED** [review]

**Pull request merged**